fbw-a380x/src/systems/systems-host/systems/powersupply.ts

```diff
@@ -0,0 +1,86 @@
+import { EventBus, EventSubscriber, Publisher, SimVarDefinition, SimVarPublisher, SimVarValueType } from '@microsoft/msfs-sdk';
+
+interface PowerSupplySimvars {
+    msfsAcBus1: number,
+    msfsAcBus2: number,
+    msfsAcBusEss: number,
+    msfsDcBus1: number,
+    msfsDcBus2: number,
+    msfsDcBusEss: number,
+}
+
+enum PowerSupplySimvarSources {
+    acBus1 = 'L:A32NX_ELEC_AC_1_BUS_IS_POWERED',
+    acBus2 = 'L:A32NX_ELEC_AC_2_BUS_IS_POWERED',
+    acBusEss = 'L:A32NX_ELEC_AC_ESS_BUS_IS_POWERED',
+    dcBus1 = 'L:A32NX_ELEC_DC_1_BUS_IS_POWERED',
+    dcBus2 = 'L:A32NX_ELEC_DC_2_BUS_IS_POWERED',
+    dcBusEss = 'L:A32NX_ELEC_DC_ESS_BUS_IS_POWERED',
+}
+
+export class PowerSupplySimvarPublisher extends SimVarPublisher<PowerSupplySimvars> {
+    private static simvars = new Map<keyof PowerSupplySimvars, SimVarDefinition>([
+        ['msfsAcBus1', { name: PowerSupplySimvarSources.acBus1, type: SimVarValueType.Number }],
+        ['msfsAcBus2', { name: PowerSupplySimvarSources.acBus2, type: SimVarValueType.Number }],
+        ['msfsAcBusEss', { name: PowerSupplySimvarSources.acBusEss, type: SimVarValueType.Number }],
+        ['msfsDcBus1', { name: PowerSupplySimvarSources.dcBus1, type: SimVarValueType.Number }],
+        ['msfsDcBus2', { name: PowerSupplySimvarSources.dcBus2, type: SimVarValueType.Number }],
+        ['msfsDcBusEss', { name: PowerSupplySimvarSources.dcBusEss, type: SimVarValueType.Number }],
+    ]);
+
+    public constructor(bus: EventBus) {
+        super(PowerSupplySimvarPublisher.simvars, bus);
+    }
+}
+
+export interface PowerSupplyBusTypes {
+    acBus1: boolean,
+    acBus2: boolean,
+    acBusEss: boolean,
+    dcBus1: boolean,
+    dcBus2: boolean,
+    dcBusEss: boolean,
+}
+
+export class PowerSupplyBusses {
```

Comment:
I don't think this class is needed. You can map the values in the simvar publisher simvar definitions. What this class ends up doing is forces all the Lvars to be read every frame even when nobody is actually subbed to the topics (the SimVarPublisher will normally not start reading anything that hasn't actually been subscribed). So, I'd suggest moving the map function (powered !== 0) to the ` PowerSupplySimvarPublisher` definitions, and plugging it into the InstrumentBackplane for the systems host instrument so it automatically gets init, startPublish, and onUpdate called.

Reply:
That's fine - I just brought it over from the A32NX and commented it out. 
I'll look into it but I did not want to touch the systems stuff itself in this PR - just provide the systems-host so that everybody can use it.

Reply:
Oh, right, fair enough I guess.